core/src/main/java/io/confluent/rest/ApplicationServer.java

```diff
@@ -128,18 +129,27 @@ public List<Application<?>> getApplications() {
     return Collections.unmodifiableList(applications);
   }
 
-  private void attachMetricsListener(Metrics metrics, Map<String, String> tags) {
-    MetricsListener metricsListener = new MetricsListener(metrics, "jetty", tags);
+  private void attachMetricsListener(String listenerName, Metrics metrics,
+      Map<String, String> tags) {
     for (NetworkTrafficServerConnector connector : connectors) {
-      connector.addNetworkTrafficListener(metricsListener);
+      if (Objects.equals(connector.getName(), listenerName)) {
+        MetricsListener metricsListener = new MetricsListener(metrics, "jetty", tags);
+        connector.addNetworkTrafficListener(metricsListener);
+        log.info("Registered {} to connector of listener: {}",
+            metricsListener.getClass().getSimpleName(), listenerName);
+      }
     }
   }
 
-  private void attachNetworkTrafficRateLimitListener() {
+  private void attachNetworkTrafficRateLimitListener(String listenerName) {
     if (config.getNetworkTrafficRateLimitEnable()) {
-      NetworkTrafficListener rateLimitListener = new RateLimitNetworkTrafficListener(config);
       for (NetworkTrafficServerConnector connector : connectors) {
-        connector.addNetworkTrafficListener(rateLimitListener);
```

Comment:
My understanding is that there would be atleast 1 listener per app, can you throw if its not the case?

Reply:
added check